source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -270,6 +275,41 @@ void AuthenticatorImpl::verifyKey() {
   handleGoodJwt(/*cache_hit=*/false);
 }
 
+void AuthenticatorImpl::addJWTClaimToHeader(const std::string& claim_name,
+                                            const std::string& header_name) {
+  StructUtils payload_getter(jwt_->payload_pb_);
+  const ::Envoy::ProtobufWkt::Value* claim_value;
+  const auto status = payload_getter.GetValue(claim_name, claim_value);
+  std::string str_claim_value;
+  if (status == StructUtils::OK) {
+    if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kStringValue) {
+      str_claim_value = claim_value->string_value();
+    } else if (claim_value->kind_case() == Envoy::ProtobufWkt::Value::kNumberValue) {
+      if (claim_value->number_value() < 0 ||
+          claim_value->number_value() >=
+              static_cast<double>(std::numeric_limits<uint64_t>::max())) {
+        // if claim is not an integer, adding a double type value in the claim.
+        headers_->addCopy(Http::LowerCaseString(header_name),
+                          std::to_string(claim_value->number_value()));
```

Comment:
write function to convert double to string,  in the function,  we can detect if it is an integer, without fraction with in correct range, then cast to int, otherwise print as double.